ext/com/concurrent_ruby/ext/SynchronizationLibrary.java

```diff
@@ -55,14 +55,14 @@ public IRubyObject initialize(ThreadContext context, IRubyObject[] args, Block b
             }
         }
 
-        @JRubyMethod(name = "synchronize", visibility = Visibility.PRIVATE)
+        @JRubyMethod(name = "synchronize", visibility = Visibility.PROTECTED)
```

Comment:
@headius Change line 58 to `Visibility.PRIVATE` and run the specs. Or simply load up IRB and use any class that subclasses `Synchronization::Object`. You will get `NoMethodError: private method `synchronize' called for...` any time a subclass calls the `#synchronize` method. Which, from where I'm sitting, is the expected behavior. Private visibility should prevent subclasses from calling a method on a parent whereas protected visibility allows subclasses to access the method. The `Synchronization::Object` class [extends](https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent/synchronization/object.rb) the `Synchronization::JavaObject` class so protected *is* the correct visibility--given the way we structure things internally.

Since `Delay` is synchronized and we use it to lazy-load the global thread pools, you'll never get the gem to load if you change line 58 above. Commit 01810052ccfc04013035ac0345d13be57cad1d2a is a quick-and-dirty update which changes the visibility of `JavaObject` and changes the visibility of `synchronize

```
$ rspec spec/concurrent/delay_spec.rb
/Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/simplecov-0.9.2/lib/simplecov.rb:31 warning: tracing (e.g. set_trace_func) will not capture all events without --debug flag
NoMethodError: private method `synchronize' called for #<Concurrent::Delay:0x4abf3f0>
     set_deref_options at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/dereferenceable.rb:87
         ns_initialize at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/delay.rb:167
            initialize at com/concurrent_ruby/ext/SynchronizationLibrary.java:54
            initialize at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/delay.rb:72
            initialize at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/utility/processor_count.rb:8
            Concurrent at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/utility/processor_count.rb:143
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/utility/processor_count.rb:4
               require at org/jruby/RubyKernel.java:1071
               require at /Users/jdantonio/.rvm/rubies/jruby-1.7.19/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/configuration.rb:1
               require at org/jruby/RubyKernel.java:1071
               require at /Users/jdantonio/.rvm/rubies/jruby-1.7.19/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent/configuration.rb:6
               require at org/jruby/RubyKernel.java:1071
               require at /Users/jdantonio/.rvm/rubies/jruby-1.7.19/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent.rb:1
               require at org/jruby/RubyKernel.java:1071
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/lib/concurrent.rb:6
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/spec/spec_helper.rb:1
                  each at org/jruby/RubyArray.java:1613
                (root) at /Users/jdantonio/Projects/opensource/concurrent-ruby/spec/spec_helper.rb:23
                (root) at /Users/jdantonio/.rvm/rubies/jruby-1.7.19/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:1
                  each at org/jruby/RubyArray.java:1613
               require at /Users/jdantonio/.rvm/rubies/jruby-1.7.19/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54
             requires= at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1181
             requires= at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1181
  process_options_into at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/configuration_options.rb:110
  process_options_into at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/configuration_options.rb:109
             configure at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/configuration_options.rb:22
                 setup at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:96
                  load at org/jruby/RubyKernel.java:1087
                (root) at /Users/jdantonio/.rvm/gems/jruby-1.7.19@concurrent-ruby/bin/rspec:23
```